### PR TITLE
backend delete + likes + joi middleware

### PR DESCRIPTION
**Pull request by marcelarie**
1. Endpoints to send likes and recive them.
2. Endpoint to delete users and songs.
3. Joi middleware with external schemas and return of 422 response+ errors when validation fails.
4. Add likes counter and almost all endpoints now work with id and name

### CODE REVIEW

File: packages/api/src/controllers/song-controller.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        if (songResponse.error) return res.status(400).send(songResponse);
+        if (!userResponse.data) return res.status(404).send(userResponse);
+        if (userResponse.data.length <= 0)
+            return res.status(204).send(userResponse);
+        if (songResponse.data)
+            return res.status(200).send({ songResponse, userResponse });
+    } catch (error) {
+        next(error);
+    }
+}
```
Comment (marcelarie) [inline packages/api/src/controllers/song-controller.js:95]: song controller method for likes

File: packages/api/src/routes/song-routes/index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+songRouter.get('/song/:name', getSongByName);
+songRouter.get('/song/id/:id', getSongById);
+songRouter.get('/song/id/:id/likes', getSongByIdWithLikes);
+songRouter.get('/songs/all', getAllSongs);
+songRouter.post('/songs/all-with', getSongsByParams);
+songRouter.post('/song/id/:id/like', likeSong);
+songRouter.post('/song', postSong);
+songRouter.patch('/song/:name', patchSongByName);
+songRouter.delete('/song/:name', deleteSong);
```
Comment (marcelarie) [inline packages/api/src/routes/song-routes/index.js:31]: songs endpoints

File: packages/api/src/routes/user-routes/index.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+const userRouter = Router();
+
+userRouter.use(authMiddleware);
+userRouter.post('/sign-up', joi(schema.signUp), signUp);
+userRouter.post('/sign-out', signOut);
+
+userRouter.get('/user/:username', getUserInfoByUsername);
+userRouter.patch('/user/edit', patchUserInfoByUsername);
+userRouter.delete('/user/delete', deleteUser);
```
Comment (marcelarie) [inline packages/api/src/routes/user-routes/index.js:23]: delete user endpoint

File: packages/api/src/middlewares/joi-middleware.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export default schema => {
+    return (req, res, next) => {
+        const { error } = schema.validate(req.body);
+        if (error) return res.status(422).send({ error });
+        next();
+    };
+};
```
Comment (marcelarie) [inline packages/api/src/middlewares/joi-middleware.js:7]: joi middleware

Comment (miguelgc95) [reply packages/api/src/middlewares/joi-middleware.js:7]: Parece muy buena idea!

**Pull request merged**